server/http_handler.go

```diff
@@ -368,7 +368,10 @@ func (t *tikvHandlerTool) getPartition(tableVal table.Table, partitionName strin
 }
 
 func (t *tikvHandlerTool) schema() (infoschema.InfoSchema, error) {
-	session, err := session.CreateSession(t.Store)
+	// Disable stats collector.
+	// Stats collector of all the sessions are linked together in the statistics handler which is a global object.
+	// If we create stats collector here, they may not be recycled by the Go GC and cause memory leak.
+	session, err := session.CreateSessionWithOpt(t.Store, &session.Opt{DisableStatsCollector: true})
```

Comment:
Seems like the root cause of the memory leak is that there is no `session.Close()` call for this `session`, so `s.statsCollector.Delete()` is not triggered to release the stats collector. Would there be other potential leaks caused by not closing this session?

Reply:
> Would there be other potential leaks caused by not closing this session?

I'm not sure. I wonder is it really necessary to create a temporary session here? Maybe that's a better fix, what's your opinion?

Reply:
Agree with @eurekaka . 
Calling `session.Close()` when exiting from these HTTP APIs seems a safer and clearer way to solve this issue.
If we add another global resource into `session` like `statsCollector`, then this problem will occur again.